File: common/src/main/java/org/tdf/common/util/RLPUtil.java
package org.tdf.common.util;


import com.github.salpadding.rlpstream.Rlp;
import com.github.salpadding.rlpstream.RlpList;
import com.github.salpadding.rlpstream.StreamId;

public class RLPUtil {
    public static <T> T decode(HexBytes data, Class<T> clazz) {
        return Rlp.decode(data.getBytes(), clazz);
    }

    public static HexBytes encode(Object o) {
        return HexBytes.fromBytes(Rlp.encode(o));
    }

    public static RlpList decodePartial(byte[] bin, int offset) {
        long streamId = StreamId.decodeElement(bin, offset, bin.length, false);
        return new RlpList(bin, streamId, 0);
    }
}

File: common/src/main/java/org/tdf/common/util/RLPUtil.kt
package org.tdf.common.util

import com.github.salpadding.rlpstream.RlpList
import com.github.salpadding.rlpstream.StreamId

object RLPUtil {
    @JvmStatic
    fun decodePartial(bin: ByteArray, offset: Int): RlpList {
        val streamId = StreamId.decodeElement(bin, offset, bin.size, false)
        return StreamId.asList(bin, streamId)
    }
}